app/models/post.rb

```diff
@@ -1,17 +1,22 @@
 class Post < ApplicationRecord
   belongs_to :user
-  validates :title, :content, :status, presence: true
   has_many :comments, dependent: :destroy
   has_many :likes, as: :likeable, dependent: :destroy
 
+  validates :title, :content, :status, presence: true
   validate :correct_file_type
   validate :file_size
+  validate :validate_published_at
 
-  enum status: { published: 0, archived: 5, draft: 10 }
+  enum status: { published: 0, archived: 5, draft: 10, scheduled: 15 }
```

Comment:
Pra restringir as mudanças para esse novo status `scheduled`, será que fica interessante sobrescrever o método `published!` ?
Acho que poderia ficar algo assim
```ruby
def published!
  if draft? || scheduled?
    super
    update(published_at: Time.zone.now)
  end
end
```
Assim só muda o status pra ```published``` e atualiza o atributo ```published_at``` se o atual for ```draft``` ou ```scheduled```

Reply:
Acho que pra lógica que vocês fizeram minha sugestão pode não fazer sentido, se for o caso podem desconsiderar, deixei o comentário antes de finalizar o review 😁

Reply:
Realmente fomos por outro caminho, mas obrigado!